plutus-core/untyped-plutus-core/src/UntypedPlutusCore/Transform/ForceDelay.hs

```diff
@@ -1,16 +1,215 @@
+{- | The 'ForceDelay' optimisation pushes 'Force' inside its direct 'Apply' subterms,
```

Comment:
Nit: we usually write long explanatory comments as Notes, rather than Haddock. I guess my general feeling is that Haddock should be useful for someone trying to use the module or functions, whereas Notes are for the implementors.